include/OpenSpaceToolkit/Astrodynamics/Flight/Profile.hpp

```diff
@@ -72,104 +72,109 @@ using ostk::astrodynamics::trajectory::State;
 class Profile
 {
    public:
-    enum class PointingMode
-    {
-
-        Undefined,  ///< Undefined pointing mode
-        Inertial,   ///< Inertial pointing mode (the spacecraft points to a celestial object)
-        Nadir,      ///< Nadir pointing mode (the spacecraft points points "directly down")
-        Target,     ///< Target pointing mode (the spacecraft points to a given target position)
-        Custom      ///< Custom pointing mode
-    };
-
+    // @brief Represents an axis of your spacecraft.
     enum class Axis
     {
         X,
         Y,
         Z
     };
 
-    enum class TargetType
+    /// @brief Represents the type of target to point towards.
+    enum class Type
     {
-        GeocentricNadir,     /// Negative of the position vector of the satellite in the ECI frame
-        GeodeticNadir,       /// Negative of the geodetic normal of the satellite in the ECI frame
-        Trajectory,          /// Points towards the provided trajectory, eg. Ground Station in ECEF
-        Sun,                 /// The position of the Sun
-        Moon,                /// The position of the Moon
-        VelocityECI,         /// The velocity vector in the ECI frame
-        VelocityECEF,        /// The velocity vector in the ECEF frame
-        OrbitalMomentum,     /// The orbital momentum vector of the satellite in the ECI frame
-        OrientationProfile,  /// Points towards a profile of orientations in the ECI frame
-        Custom,              /// Custom target
+        // Simple pre-defined targets
+        GeocentricNadir,  /// Negative of the position vector of the satellite in the ECI frame
+        GeodeticNadir,    /// Negative of the geodetic normal of the satellite in the ECI frame
+        Sun,              /// The position of the Sun
+        Moon,             /// The position of the Moon
+        VelocityECI,      /// The velocity vector in the ECI frame
+        VelocityECEF,     /// The velocity vector in the ECEF frame
+        OrbitalMomentum,  /// The orbital momentum vector of the satellite in the ECI frame
+
+        // More complex targets
+        Trajectory,  /// Points towards the provided trajectory, eg. Ground Station in ECEF or another satellite
+
+        // Custom pointing
+        CustomProfileECI,           /// Pointing pre-defined via a profile of vectors in the ECI frame
+        CustomProfileGeneratorECI,  /// Pointing generated at each timestep by a function of that returns a vector in
+                                    /// the ECI frame
```

Comment:
```suggestion
        Trajectory,  /// Points towards the provided trajectory, eg. Ground Station in ECEF or another satellite
        Custom /// Pointing generated at each timestep by a function that returns a Quaternion in the ECI frame
```
Not sure if we want 2 different custom profiles here? They're both just technically generators?

Reply:
Ah good point